src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.CoreCLR.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Diagnostics;
+using System.Runtime.Versioning;
+
+namespace System.Collections.Generic
+{
+    internal interface IArraySortHelper<TKey>
+    {
+        void Sort(TKey[] keys, int index, int length, IComparer<TKey> comparer);
+        int BinarySearch(TKey[] keys, int index, int length, TKey value, IComparer<TKey> comparer);
+    }
+
+    [TypeDependencyAttribute("System.Collections.Generic.GenericArraySortHelper`1")]
+    internal partial class ArraySortHelper<T>
+        : IArraySortHelper<T>
+    {
+        private static readonly IArraySortHelper<T> defaultArraySortHelper = CreateArraySortHelper();
```

Comment:
Please rename defaultArraySortHelper to s_defaultArraySortHelper since it also has this s_ name in corert

Reply:
Done